caluma/caluma_workflow/migrations/0020_case_families.py

```diff
@@ -0,0 +1,71 @@
+# Generated by Django 2.2.10 on 2020-03-05 09:17
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def set_family(apps, schema_editor):
+    Case = apps.get_model("caluma_workflow", "Case")
+    HistoricalCase = apps.get_model("caluma_workflow", "HistoricalCase")
+    db_alias = schema_editor.connection.alias
+
+    orphan_cases = Case.objects.using(db_alias).filter(family__isnull=True)
+    orphan_historical_cases = HistoricalCase.objects.using(db_alias).filter(
+        family__isnull=True
+    )
+
+    for case in orphan_cases:
+        family = case
+        while hasattr(family, "parent_work_item"):
```

Comment:
Out of interest, why `while` and not `if`?

Reply:
@sbor23 Because you want the family to point to the top-level case, not just the parent, so you need to go as far up the tree as you can

Reply:
Ah gotcha!